hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/metadata/SparkHoodieBackedTableMetadataWriter.java

```diff
@@ -144,6 +144,9 @@ protected void commit(HoodieData<HoodieRecord> hoodieDataRecords, String partiti
         metadataMetaClient.reloadActiveTimeline();
       }
       List<WriteStatus> statuses = writeClient.upsertPreppedRecords(recordRDD, instantTime).collect();
+      if (canTriggerTableService) {
```

Comment:
wondering if we should move this to after 161. to be specific, after compaction and cleaning. Even in regular flow, cleaning comes first followed by archival.